src/services/fireblocks.service.js

```diff
@@ -148,7 +158,19 @@ async function getTxSignatures(tx, storage, client) {
     if (!sig) {
       throw new Error(`Fireblocks did not return signature for address index ${ncCallerIndex} in transaction(${fbTxInfo.id}); external-id: (${dataToSignHash.toString('hex')})`);
     }
-    ncCallerSignature = fireblocksSignatureToDER(sig.signature);
+    const pubkeyHex = await storage.getAddressPubkey(ncCallerIndex);
+    if (sig.publicKey !== pubkeyHex) {
+      throw new Error(`Fireblocks signature does not match locally generated public key in transaction(${fbTxInfo.id}); external-id: (${dataToSignHash.toString('hex')})`);
+    }
```

Comment:
suggestion(non-blocking): this could be moved inside the `if (tx.isNanoContract())`

Reply:
Since this validation works for both the if and else blocks I think it should remain outside the if. what do you think.

Reply:
It's ok. You don't use `sig.publicKey` but you can validate it.